src/plugins/workspace/server/saved_objects/workspace_saved_objects_client_wrapper.ts

```diff
@@ -330,6 +343,9 @@ export class WorkspaceSavedObjectsClientWrapper {
       attributes: T,
       options?: SavedObjectsCreateOptions
     ) => {
+      // Only OSD admin can create workspace
+      if (type === WORKSPACE_TYPE) throw generateOSDAdminPermissionError();
```

Comment:
https://github.com/opensearch-project/OpenSearch-Dashboards/blob/d911fa7144fec19b9409a8dea6b7d4d06af9a2b8/src/plugins/workspace/server/workspace_client.ts#L195

It seems we are using `create with overwrite` to `update` the workspace object.

Reply:
Yes, it's by intention. Here is how update api behaves, what we want to update is to fully replace existing document. 

> The update API also supports passing a partial document, which is merged into the existing document. To fully replace an existing document, use the [index API].

Reply:
Make sense, when users update a workspace, `createWithWorkspacePermissionControl` with `options.overwrite=true` will be called. 
The code of checking overwrite has been added.

Reply:
Call update API with permissions object won't remove existing permissions attributes. For example, there is a workspace saved object with permissions attribute like:
```
{
    read: {users:['foo']}
}
```
If I call update API with 
```
{
    library_read: {users:['foo']}
}
```
The read permission won't be updated by savedObjects.update method. That's why we use create method here.